libraries/AP_Scripting/applets/set_home_position.lua

```diff
@@ -0,0 +1,41 @@
+-- Script to set new home position with the RC
+-- Sets the preset or current home position after 2 seconds that the switch is in the chosen position
+-- Author: Marco Robustini
+
+local rc_option = 300
+local channel = rc:find_channel_for_option(rc_option)
+  if not channel then
+return
+local sw_state = -1
+
+-- Define and initialize the location object with desired coordinates; enter the coordinates with the first 9 numbers and the height in centimetres
+my_home_location = Location()
+my_home_location:lat(448060084)
+my_home_location:lng(116100275)
+my_home_location:alt(0000)
+
+function update()
+  local sw_pos = channel:get_aux_switch_pos()
+  if sw_state ~= sw_pos or sw_pos == 2 then
+    if sw_pos == 1 then
+      ahrs:set_home(my_home_location)
+      gcs:send_text(6, string.format("Set Home to preset position: Lat:%.7f Long:%.7f Alt:%.1f", my_home_location:lat()/10000000, my_home_location:lng()/10000000, my_home_location:alt()/100))
+	  gcs:send_text(1, string.format("Home position updated"))
+    elseif sw_pos == 2 then
+      if ahrs:home_is_set() then
+        local location = ahrs:get_location()
+        if location then
+          location:alt(0)
```

Comment:
I think this is in abs frame? So 0 will be underground? You would have to get home and copy its altitude.

Reply:
@IamPete1 you mean get current home and assume new home is at same altitude?
How would we go about getting the actual Z clearance below the vehicle, we would need to use terrain I guess

Reply:
Yes @IamPete1, abs frame.
I have to extract the original home height but I haven't figured out how, so I can use that.
Then there would also be the variable 'terrain' to take into account, too complex for my skills.

Reply:
I think just using the height of current home is better than nothing, If you want to use terrain you can use `terrain:height_amsl(location, true)`.

Reply:
And how to set the original home height instead of "location:alt(0)"?

Reply:
> And how to set the original home height instead of "location:alt(0)"?
before setting the new home you save the current one and "copy" the altitude from old to new, should work